c7n/resources/cloudfront.py

```diff
@@ -181,6 +183,66 @@ def process(self, resources, event=None):
         return results
 
 
+@Distribution.filter_registry.register('check-s3-origin')
+class CheckS3Origin(Filter):
```

Comment:
Rather than inherit from Filter, why not ValueFilter to get the whitelist functionality out of the box?